crates/typst/src/syntax/ast.rs

```diff
@@ -841,6 +845,36 @@ impl MathAttach {
             .skip_while(|node| !matches!(node.kind(), SyntaxKind::Hat))
             .find_map(SyntaxNode::cast)
     }
+
+    pub fn has_primes(&self) -> bool {
+        self.0.children().any(|node| matches!(node.kind(), SyntaxKind::Prime))
+    }
+
+    /// Extract primes
+    pub fn primes(&self) -> MathPrimes {
+        MathPrimes(SyntaxNode::inner(
+            SyntaxKind::MathPrimes,
```

Comment:
constructing the SyntaxNode ad-hoc here is not ideal, it would not have a span. either wrap the primes in MathPrimes in the parser (I think you already do?) or don't and handle the primes fully in the attach node, removing `MathPrimes`. in that case `has_primes` could be replaced by `primes(&self) -> Option<usize>`.